server/cache/lru_ttl_cache.go

```diff
@@ -0,0 +1,47 @@
+package cache
+
+import (
+	"time"
+
+	"k8s.io/utils/lru"
+)
+
+type lruTtlCache struct {
+	timeout time.Duration
+	cache   *lru.Cache
+}
+
+type timeValueHolder struct {
+	createTime time.Time
+	value      any
```

Comment:
minor - go full generics

```golang
type item[T] struct  {
creationTime time.Time
item T
}
```

Reply:
I had done that,
The problem is there is a bug while doing go generate for mockery
https://github.com/golang/go/issues/51629
Had to revert the generics code

Reply:
ah